efel/settings.py

```diff
@@ -19,16 +19,135 @@
+        rise_start_perc (float): Rise start percentage (default: 0.0).
+        rise_end_perc (float): Rise end percentage (default: 1.0).
+        initial_perc (float): Initial percentage (default: 0.1).
+        min_spike_height (float): Minimum spike height (default: 20.0).
+        strict_stiminterval (int): Strict stimulus interval (default: 0).
+        initburst_freq_threshold (int): Initial burst frequency threshold
+        (default: 50)
+        initburst_sahp_start (int): Initial burst SAHP start (default: 5).
+        initburst_sahp_end (int): Initial burst SAHP end (default: 100).
+        DerivativeWindow (int): Derivative window (default: 3).
+        voltage_base_mode (str): Voltage base mode (default: "mean").
+        current_base_mode (str): Current base mode (default: "mean").
+        precision_threshold (float): Precision threshold (default: 1e-10).
+        sahp_start (float): SAHP start (default: 5.0).
+        ignore_first_ISI (int): Ignore first ISI (default: 1).
+        impedance_max_freq (float): Impedance maximum frequency (default: 50.0).
+    """
+
+    Threshold: float = -20.0
+    DerivativeThreshold: float = 10.0
+    DownDerivativeThreshold: float = -12.0
+    dependencyfile_path: str = str(
+        Path(__file__).parent.absolute() / "DependencyV5.txt"
+    )
+
+    spike_skipf: float = 0.1
+    max_spike_skip: int = 2
+    interp_step: float = 0.1
+    burst_factor: float = 1.5
+    strict_burst_factor: float = 2.0
+    voltage_base_start_perc: float = 0.9
+    voltage_base_end_perc: float = 1.0
+    current_base_start_perc: float = 0.9
+    current_base_end_perc: float = 1.0
+    rise_start_perc: float = 0.0
+    rise_end_perc: float = 1.0
+    initial_perc: float = 0.1
+    min_spike_height: float = 20.0
+    strict_stiminterval: int = 0
+    initburst_freq_threshold: int = 50
+    initburst_sahp_start: int = 5
+    initburst_sahp_end: int = 100
+    DerivativeWindow: int = 3
+    voltage_base_mode: str = "mean"
+    current_base_mode: str = "mean"
+    precision_threshold: float = 1e-10
+    sahp_start: float = 5.0
+    ignore_first_ISI: int = 1
+    impedance_max_freq: float = 50.0
+
+    def set_setting(self, setting_name: str, new_value: Union[int, float, str]) -> None:
+        """Set a certain setting to a new value.
+
+        Args:
+            setting_name (str): Name of the setting to be modified.
+            new_value (Union[int, float, str]): New value for the setting.
+
+        Raises:
+            ValueError: If the value is of the wrong type.
+            FileNotFoundError: If the path to the dependency file does not exist
+            (for 'dependencyfile_path' setting).
+        """
+        if hasattr(self, setting_name):
+            expected_types = {f.name: f.type for f in fields(self)}
+            expected_type = expected_types.get(setting_name)
+            if expected_type and expected_type != type(new_value):
```

Comment:
One comment, if you use `isinstance` here, users can keep passing bool values for int. 
```
>>> isinstance(True, int)
True
>>> type(True) == type(int)
False
>>> type(True)
<class 'bool'>
>>> type(int)
<class 'type'>

```